Jumpscale/builder/apps/BuilderFreeflow.py

```diff
@@ -1,142 +1,64 @@
 from Jumpscale import j
 import textwrap
 
+builder_method = j.builder.system.builder_method
+
 
 class BuilderFreeflow(j.builder.system._BaseClass):
     NAME = "freeflow"
-
-    def _init(self):
-
+    @builder_method()
+    def install(self, reset =False):
         self.HUMHUB_PATH = "/var/www/html/humhub"
-        # self.dirs = {
-        #     "/var/www/html/humhub": "/var/www/html/humhub",
-        #     "/var/lib/mysql/": "/var/lib/mysql/",
-        #     "/var/log/mysql/": "/var/log/mysql/",
-        #     "/var/run/mysqld/": "/var/run/mysqld/",
-        #     "/etc/apache2": "/etc/apache2"
-
-        # }
-        # self.dirs.update(j.builder.db.mariadb.dirs)
-
-        # self.bins = [
-        #     "/usr/bin/mysql",
-        #     "/usr/sbin/apache2"
-        # ]
-        # self.bins.extend(j.builder.db.mariadb.bins)
-
-        self.new_files = dict()
-        self.new_files.update(j.builder.db.mariadb.new_files)
-
-        # startup_file = j.sal.fs.joinPaths(j.sal.fs.getDirName(__file__), 'templates', 'freeflow_startup.toml')
-        # self.startup = j.sal.fs.readFile(startup_file)
-
-
-    def build(self, reset =False):
-
-        if self._done_check("build") and reset is False:
+        if self._done_check("install") and reset is False:
             return
-        j.builder.db.mariadb.install(start=True)
-        j.builder.system.package.install(["apache2",
+        #j.builder.db.mariadb.install(start=True)
+        # i remove lamp server as it waste time in testing "lamp-server^"
+        j.builder.system.package.install(["lamp-server^",
                                          "php-curl",
                                          "php-gd",
                                          "php-mbstring",
                                          "php-intl",
                                          "php-zip",
                                          "php-ldap",
-                                         "php-apcu"
+                                         "php-apcu",
+                                         "php-sqlite3",
+                                          "php-imagick"
                                          ])
 
-        j.builder.tools.file_download("https://www.humhub.org/en/download/package/humhub-1.3.7.tar.gz",
-                                      "/var/www/html/humhub-1.3.7.tar.gz")
-        j.builder.tools.file_expand("/var/www/html/humhub-1.3.7.tar.gz", self.HUMHUB_PATH, removeTopDir=True)
+        #j.builder.tools.file_download("https://www.humhub.org/en/download/package/humhub-1.3.12.tar.gz","/var/www/html/humhub-1.3.12.tar.gz")
```

Comment:
why this line is commented out how would you download, I know it will raise an error if you have the file already downloaded but you need to check for it instead of commenting the line out